### ISSUE TITLE
Can't install on Spring Tool Suite

### ISSUE THREAD

**Issue by aroneous** [groovy/groovy-eclipse#284]
I've been unable to install Groovy-Eclipse on Spring Tool Suite at least since the beginning of the year. I've tried on release 3.8.3 and now on current 3.8.4,  but always get a dependency conflict. This happens on a clean install of STS, with the Groovy-Eclipse install being the first thing I attempt.

I've tried to make sense of what the conflict actually is, but don't really understand it. Interestingly, I have an existing STS install with Groovy-Eclipse installed long ago, and it's able to update both Groovy-Eclipse and STS versions without issue. Not sure if this is a Groovy-Eclipse problem per se, but I'm hoping somebody might have some insight.

I'll attach the error message separately. This is on Windows x64 version of STS.

**Comment by aroneous** [comment]
Dependency conflict error:

[groovy-eclipse-conflict.txt](https://github.com/groovy/groovy-eclipse/files/948140/groovy-eclipse-conflict.txt)

**Comment by eric-milles** [comment]
I was helping someone on the slack channel with this earlier in the week.  The long and short of it is that Groovy-Eclipse needs to patch a JDT Core bundle.  The feature Object Teams Development Tooling Core tries to patch the same bundle.  This is not allowed and so the installation process tries to explain this as best as possible.

I'm not sure if STS comes with Object Teams Development Tooling pre-installed.  If you can somehow get it to uninstall from Help > Install New Software... > Installation Details dialog, Groovy-Eclipse should install cleanly.

**Comment by eric-milles** [comment]
What did you start with in terms of Eclipse?  I use the Platform Runtime Binary because it has the minimal set of stuff.  When you added STS to that, what features did you check?  There are a lot of choices given by the STS update site.

**Comment by aroneous** [comment]
I installed the Spring-curated bundling of Eclipse: http://spring.io/tools/sts
I.e., I didn't install Eclipse first and install Spring plugins. There is, so far as I know, only one variant of the package.

I went looking for Object Teams Development Tooling in my install, but I've been unable to find it. The version of org.eclipse.jdt.core that I have is 3.12.3.v20170228-1205, which doesn't seem to be one of the Object Teams variants mentioned. Hence my confusion...

**Comment by eric-milles** [comment]
Where is the download you used?  I do not see one on the page you have linked.  When I searched STS on Google, I went to the download page and received an update site archive.  This I was able to add to my eclipse and then install Groovy-Eclipse with no issue.

![sts](https://cloud.githubusercontent.com/assets/18193802/25294365/6efccb6a-26a4-11e7-8835-c2bfdc8193dc.png)

**Comment by aroneous** [comment]
Strange - that page for me has a big button on the right that says 'DOWNLOAD STS (3.8.4.RELEASE for Windows)'. It links to http://download.springsource.com/release/STS/3.8.4.RELEASE/dist/e4.6/spring-tool-suite-3.8.4.RELEASE-e4.6.3-win32-x86_64.zip

There's a 'See All Versions' link that points here: http://spring.io/tools/sts/all

**Comment by eric-milles** [comment]
Okay, I see that button now.  Hidden in plain sight.  When I unpack the bundle and add Groovy-Eclipse I get no conflict.

**Comment by aclement** [comment]
I just installed greclipse from ( http://dist.springsource.org/snapshot/GRECLIPSE/e4.7/ ) into my downloaded MacOS STS 3.8.4 (the oxygen version) and it worked fine.  Yes we would clash with OTDT but we don't include OTDT in STS.

**Comment by aroneous** [comment]
Stranger and stranger... Maybe it's something environmental; I'll have to keep digging. Thanks for your efforts.

**Comment by aclement** [comment]
As an extra check I just downloaded STS 3.8.3 for windows (neon.3 based) - installed the greclipse snapshot for e4.6 into it. It worked ok. I only chose the main greclipse feature and the 2.4 compiler of all the stuff on the update site.

**Comment by davidmichaelkarr** [comment]
I'm the slacker Eric was helping earlier.  I had originally installed the "plain Java" edition from the installer, but I've been updating it in place for probably the last two major releases.  I'm up to Neon.3 now.

I first saw this a few weeks ago when I tried to update, and reported this same conflict.  To try to get past this, I uninstalled both groovy-eclipse and spring ide, then installed groovy-eclipse, and then spring ide.  That worked, although it took a while to figure that out.

After that, I tried to update again, and it reported that in order to proceed, the "Will be uninstalled" list was all of the Spring IDE features.  I let it do that, then restarted, and then installed Spring IDE again.  That completed.  I think I'm going to have to do this every time I update, and I hope it continues to work.
